ftw/iframeblock/browser/resources/iframeblock.js

```diff
@@ -1,9 +1,33 @@
-(function() {
-
+(function (factory) {
+  if (typeof define === 'function' && define.amd) {
+    require(['jquery', 'iframeResizer'], factory);
+  } else {
+    factory(window.jQuery, window.iFrameResize);
+  }
+}(function ($, iFrameResize) {
   "use strict";
 
-  window.onIframeLoaded = function(element) {
-      $(element).prev().removeClass('loading');
-  };
+  function initIFrame() {
+      var iframe = $(this);
+
+      if (iframe.data('autoSize') === 'True') {
+        iFrameResize({
+          inPageLinks: true,
+          heightCalculationMethod: iframe.data('heightCalculationMethod'),
+          onResized: function () {scroll(0, 0);},
+        }, this);
+      }
+
+      iframe.prev().removeClass('loading');
+  }
+
+  // We have to wait for the ready and then add the onload event listener directly on the
+  // iframe because the iframe does not propagate the onload back to document.
+  $(document).ready(function () {
+    $('iframe').on('load', initIFrame);
 
-})();
+    $(document).on('onBeforeClose', '.overlay', function () {
```

Comment:
I think 'onClose' is probably a little better, but it's "small potatoes" - I'll let you decide

Reply:
According to the `ftw.simplelayout` docs one should use this.
https://github.com/4teamwork/ftw.simplelayout/#run-js-after-adding-and-editing-a-block

Reply:
Ok, maybe they want it looking good before the overlay disappears?